sort.c

```diff
@@ -45,34 +54,40 @@ static void replySortedArray(RedisModuleCtx *ctx, RedisModuleCallReply * reply,
         case REDISMODULE_REPLY_SET:
             RedisModule_ReplyWithSet(ctx, len);
             break;
+        default:
+            /* shouldn't get here, as should be protected above */
+            assert(0);
```

Comment:
Better use `RedisModule_Assert` (and drop `assert.h`).

Reply:
1) oh, I see.  trick of storing the value in the pointer.  also, that I was using Set wrong, it only worked as an artifact of the fact that I was getting the pointer and incrementing it.  with that said, that is an optimization in a way (the only Q would be, which is less efficient, walking the tree an extra time per key, or allocating memory. I guess allocating memory.

2) ok, I think I understand that

3).Easy enough change